controllers/openshift_controller.go

```diff
@@ -277,7 +278,18 @@ func (r *KataConfigOpenShiftReconciler) processDaemonsetForMonitor() *appsv1.Dae
 		runGroupID    = int64(1001)
 	)
 
-	r.Log.Info("Creating monitor DaemonSet with image file: " + r.kataConfig.Spec.KataMonitorImage)
+	kataMonitorImage := os.Getenv("KATA_MONITOR_IMAGE")
+	if len(kataMonitorImage) == 0 {
+		// kata-monitor image URL is generally impossible to verify or sanitise,
+		// with the empty value being pretty much the only exception where it's
+		// fairly clear what good it is.  If we can only detect a single one
+		// out of an infinite number of bad values, we choose not to return an
+		// error here (giving an impression that we can actually detect errors)
+		// but just log this incident and plow ahead.
+		r.Log.Info("KATA_MONITOR_IMAGE env var is unset or empty, kata-monitor pods will not run")
```

Comment:
love this explanation :-)

Reply:
Well you can see I really wanted to find a way to return an error that made sense but couldn't. ;-)

The decision isn't clear cut so to prevent someone (probably myself) thinking three months down the line, "what the hell did they think they were doing?", I thought I'd document it.